operator/pkg/ztunnel/ztunnel-daemonset.yaml

```diff
@@ -0,0 +1,139 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: ztunnel-cilium
+  labels:
+    app: ztunnel
+    app.kubernetes.io/name: ztunnel
+    app.kubernetes.io/managed-by: cilium-operator
+spec:
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: ztunnel
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+      labels:
+        app: ztunnel
+    spec:
+      hostNetwork: true
+      containers:
+      - args:
+        - proxy
+        - ztunnel
+        env:
+        - name: XDS_ADDRESS
+          value: "https://localhost:15012"
+        - name: XDS_ROOT_CA
+          value: "/etc/ztunnel/bootstrap-root.crt"
+        - name: CA_ROOT_CA
+          value: "/etc/ztunnel/bootstrap-root.crt"
+        - name: CA_ADDRESS
+          value: "https://localhost:15012"
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: spec.nodeName
+        - name: INSTANCE_IP
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: status.podIP
+        - name: SERVICE_ACCOUNT
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: spec.serviceAccountName
+        - name: ISTIO_META_ENABLE_HBONE
+          value: "true"
+        image: docker.io/istio/ztunnel:1.27.0-distroless
```

Comment:
Should this be configurable/overridable?

Reply:
No, sounds good. I would suggest we use the latest release though, e.g. 1.27.3 or 1.28.0